kie-api/src/main/java/org/kie/api/runtime/rule/QueryResultsRow.java

```diff
@@ -21,4 +21,18 @@
  */
 public interface QueryResultsRow extends Row {
 
+    /**
+     * Return the Object for the given Declaration identifier.
+     * @param identifier
+     * @return the Object
+     */
+    public Object get(String identifier);
+
+    /**
+     * Return the {@link FactHandle} for the given Declaration identifier.
+     * @param identifier
+     * @return the {@link FactHandle} instance
+     */
+    public FactHandle getFactHandle(String identifier);
+
```

Comment:
Correct: 1. end-users should _not_ be implementing this interface and 2. adding new methods willnot (AFAIK) break backwards compatibility. The underlying implementations already were implementing both of these methods.